scripts/configure-venue-access.ts

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+
+import fs from "fs";
+import admin from "firebase-admin";
+import "firebase/firestore";
+
+import {
+  VenueAccessCodes,
+  VenueAccessEmails,
+  VenueAccessMode,
+} from "../src/types/VenueAcccess";
+
+import { initFirebaseAdminApp } from "./lib/helpers";
+
+const mergeStringArrays = (
+  arr1: string[] = [],
+  arr2: string[] = []
+): string[] => {
+  const result = arr1.map((val) => val.trim());
+  arr2.forEach((val) => {
+    const valTrimmed = val.trim();
+    if (!result.includes(valTrimmed)) {
+      result.push(valTrimmed);
+    }
+  });
+  return result;
+};
+
+function usage() {
+  console.log(`
+${process.argv[1]}: Configure venue access. Supports configuring a secret password, list of emails which can access the venue, or ticket codes.
+
+Usage: npx ts-node ${process.argv[1]} PROJECT_ID VENUE_ID [password|emaillist|codelist] [password | emails file path | codes file path]
+
+Example: npx ts-node ${process.argv[1]} co-reality-map password abc123
+Example: npx ts-node ${process.argv[1]} co-reality-map emails emails-one-per-line.txt
+Example: npx ts-node ${process.argv[1]} co-reality-map codes ticket-codes-one-per-line.txt
+`);
+  process.exit(1);
+}
+
+const argv = process.argv.slice(2);
+if (argv.length < 4) {
+  usage();
+}
+
+const projectId = argv[0];
+const venueId = argv[1];
+const method = argv[2];
+const accessDetail = argv[3];
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+if (!(Object as any).values(VenueAccessMode).includes(method)) {
+  console.error(`Invalid access method ${method}`);
+  process.exit(1);
+}
+
+initFirebaseAdminApp(projectId)(async () => {
+  console.log(`Ensuring ${venueId} access via ${method} - ${accessDetail}`);
+  const venueDoc = await admin.firestore().doc(`venues/${venueId}`).get();
```

Comment:
Why not use this shorter syntax and change the rest of the code? When there is a subcollection the chain becomes too long.

Reply:
In this instance, I don’t have particularly strong reasoning for/against; aside from it being more explicit when not joined as one ‘path’.

if this wasn’t an engineering script, and these values were coming from potentially user controllable data, I would be concerned that this ‘path’ way would be vulnerable to essentially SQL injection, by manipulating the path to access a different document than was originally intended. My feeling is that by using the more explicit method we would avoid that.

The other element of my reasoning is that most examples I’ve seen both within our codebase currently + in the docs seem to use the more explicit version, rather than this path version.

Reply:
Backlogged this general issue in our codebase in https://github.com/sparkletown/internal-sparkle-issues/issues/248